### PR TITLE
Change use of world_to_pixel / pixel_to_world to wcs.world_to_pixel / pixel_to_world in the docs.

### PR DESCRIPTION
**Pull request by ryuusama09**
## PR Description
 fixes wcs in codebase except tests.
supersedes pr : https://github.com/sunpy/sunpy/pull/6723
fixes : https://github.com/sunpy/sunpy/issues/6435

### CODE REVIEW

File: CHANGELOG.rst
Comment (nabobalis) [inline CHANGELOG.rst:483]: This file needs to be reset back to normal.

File: docs/intro/coordinates.rst
Comment (nabobalis) [inline docs/intro/coordinates.rst:95]: This output will now be different.

File: docs/intro/coordinates.rst
Comment (nabobalis) [inline docs/intro/coordinates.rst:91]:
```suggestion
   The `~astropy.coordinates.SkyCoord` object can be converted to a pair of pixels using `~sunpy.map.GenericMap.wcs.world_to_pixel`:
```

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:119]: This file should fail as far as I can see.

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:47]: This example should also fail if I understand correctly.

**Comment by nabobalis** [comment]
> @nabobalis , i dont know how the map tests are failing , theyre just working fine locally 😥

When the tests run, I will check the logs.

**Comment by nabobalis** [comment]
Looks like you need to change:

```
>       xy2 = np.dstack([x2.T.value.flat, y2.T.value.flat])[0]
```
to account for the change in world to pixel output.

File: examples/plotting/xy_lims.py
Comment (nabobalis) [reply examples/plotting/xy_lims.py:47]: > ill check on this . also why does changelog needs to be reset. It should show that we have changed to wcs right

That is not how we do that. The file should be changed back to normal.

Comment (nabobalis) [reply examples/plotting/xy_lims.py:47]: Sure, but now we don't have a pixel pair, you have to change the lines in th example that assume it is a pixel pair.

Comment (nabobalis) [reply examples/plotting/xy_lims.py:47]: No, but we don't need that. We can change the other lines in the example.

Comment (nabobalis) [reply examples/plotting/xy_lims.py:52]: We should not be using PixelPair, you need to change the other lines of the example.

The changelog file still has changes.

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:484]: Since this has changed, the places where x2 and y2 are used, need to be updated.

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:49]:
```suggestion
```
You need to update the lines below.

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [reply sunpy/physics/differential_rotation.py:484]: Something has to be changed to make the code work again. If x2 and y2 here arent, then what should we change?

Comment (nabobalis) [reply sunpy/physics/differential_rotation.py:484]: world_to_pixel always returns two values. That can either be assigned to one variable or two, like in this case.

We don't need to worry about pixel_to_world

Comment (nabobalis) [reply sunpy/physics/differential_rotation.py]: The changelog has not been reverted.

The fix can not be using pixelpair, we have to change other parts of the code.

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:17]: We can not use this to fix the example, the answer lies in changing other lines of code.

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:8]:
```suggestion
```

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:48]:
```suggestion
```

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:17]:
```suggestion
```

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:112]:
```suggestion
```

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:116]:
```suggestion
xpix, ypix = magnetogram_big.wcs.world_to_pixel(right_corner)
```

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:118]:
```suggestion
    (0, 1), (xpix, ypix), 'axes fraction', 'data', axesA=ax2, axesB=ax1,
```

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:121]:
```suggestion
xpix, ypix = magnetogram_big.wcs.world_to_pixel(
```

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:125]:
```suggestion
```

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:127]:
```suggestion
    (0, 0), (xpix, ypix), 'axes fraction', 'data', axesA=ax2, axesB=ax1,
```

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:51]:
```suggestion
```

Comment (nabobalis) [reply examples/plotting/xy_lims.py:51]: The line below has to go as well.

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:488]:
```suggestion
        x2, y2 = smap.wcs.world_to_pixel(coordinates_at_map_observer)
```

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:492]:
```suggestion
    xy2 = np.dstack([x2.T.flat, y2.T.flat])[0]
```

**Comment by nabobalis** [comment]
> I will revert the changelog. 

Thanks

> But it is necessary to use pixelpair because x2 ,y2 have to be a pixel quantity. To make x2,y2 as a pixel quantity we need pixelpair.

Why do x2 and y2 need to be pixels?

File: examples/plotting/hmi_cutout.py
Comment (nabobalis) [inline examples/plotting/hmi_cutout.py:21]:
```suggestion
```

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:16]:
```suggestion
import sunpy.map

```

File: examples/plotting/xy_lims.py
Comment (nabobalis) [reply examples/plotting/xy_lims.py:48]: Why do they need to be a pixel pair?

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:7]:
```suggestion
from astropy.time import TimeDelta

```

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:453]:
```suggestion
        map_coord = smap.wcs.pixel_to_world(*input_pixels)
```

**Comment by nabobalis** [comment]
> > Why do x2 and y2 need to be pixels?
> 
> because during tests it mentioned , "numpy.ndarray has not attribute value" . this meant that it has to be a pixel quantity.

Ok, so why do we not change this instead of making them a pixel pair?

**Comment by nabobalis** [comment]
> @nabobalis , only the `changelog.rst` has to be changed or all files in changelog directory ?

only the files you changed.

File: CHANGELOG.rst
Comment (nabobalis) [inline CHANGELOG.rst:929]: These should not be here.

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:462]:
```suggestion
```

File: sunpy/physics/differential_rotation.py
Comment (nabobalis) [inline sunpy/physics/differential_rotation.py:486]:
```suggestion
```

**Comment by nabobalis** [comment]
> then i think we will have to change the line ` xy2 = np.dstack([x2.T.value.flat, y2.T.value.flat])[0]` to something else.

I think we should do that.

> I personally feel we should keep it as a pixel quantity as it will show how to use `wcs.wtp` to derive a pixel quantity and not a tuple

This code is internal, I don't understand how it would do that.

File: CHANGELOG.rst
Comment (nabobalis) [reply CHANGELOG.rst:929]: That is there in your code base, the `wcs` is missing.

**Comment by nabobalis** [comment]
> then i guess it would simply mean `np.dstack(x2,y2)`.

So far the tests are passing with the current version.

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:46]:
```suggestion
```

File: examples/plotting/xy_lims.py
Comment (nabobalis) [inline examples/plotting/xy_lims.py:44]: You will need to update the lines below that set the xlim and ylim otherwise this example will error.

Comment (nabobalis) [reply examples/plotting/xy_lims.py:44]: That is surprising to me, if that's the case, then that's ok.

**Comment by nabobalis** [comment]
That looks like everything @ryuusama09.
Thanks for the powering through the longer review process for this issue.

**Comment by nabobalis** [comment]
> i think we are good to go for merge ? @nabobalis

This will need a review from others first.

**Comment by nabobalis** [comment]
> @nabobalis , is there any issue over here that it isnt being merged?

No, I have requested a review from another person and we will wait for that before this gets merged.

**Comment by nabobalis** [comment]
Can you merge in the latest version of sunpy/main?

I want to check all the tests pass.

**Comment by nabobalis** [comment]
Thanks

**Comment by nabobalis** [comment]
If you could but I think there is also a genuine issue that needs fixing in that pull request.

**Comment by nabobalis** [comment]
> @nabobalis , i would like to connect with you on mail regarding the gsoc 2023 projects . If possible could ou share your email ?

Please do not @ people unless it's important. 

All GSoC communications should happen in the main chat room: https://app.element.io/#/room/#sunpy:openastronomy.org 

If you do still need to email me: nabil.freij@gmail.com

File: sunpy/map/maputils.py
Comment (wtbarnes) [inline sunpy/map/maputils.py:164]:
```suggestion
    edge_pixels = np.concatenate(edges).value
```

File: sunpy/physics/differential_rotation.py
Comment (wtbarnes) [inline sunpy/physics/differential_rotation.py:340]:
```suggestion
    c = smap.wcs.pixel_to_world(pixels[:, 0].value, pixels[:, 1].value)
```

File: sunpy/physics/differential_rotation.py
Comment (wtbarnes) [inline sunpy/physics/differential_rotation.py:453]:
```suggestion
        input_pixels = xy.T
```

**Review by wtbarnes — CHANGES_REQUESTED** [review]
I've taken the approach here of making any input to `wcs.world_to_pixel` unit-less when it is used *internally*. Externally, I've made no changes as I don't think that is within scope of this PR.

**Review by dstansby — APPROVED** [review]

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks again @ryuusama09

**Pull request merged**
